### PR TITLE
Fix nxp 21435 clean chronicle queues after import

### PR DESCRIPTION
**Pull request by dmetzler**

### CODE REVIEW

**Comment by bdelbosc** [comment]
:+1:

File: nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/CQManager.java [hunk 1]
```diff
@@ -78,8 +78,22 @@ public void init() {
                 tailers.add(queue.createTailer().toEnd());
             }
         }
+        isSetup = true;
     }
 
+    @Override
+    public void stop() {
+        try {
+            queues.stream().forEach(ChronicleQueue::close);
```
Comment (efge) [inline nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/CQManager.java]: You can skip the `stream()` part and directly call `.forEach` on any `Iterable`.

File: nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/QueueImporter.java [hunk 2]
```diff
@@ -102,6 +103,9 @@ public void importDocuments(List<Producer> producers, QueuesManager manager, Str
             isRunning = false;
         }
 
+        // Cleans the queuing mecanism if needed
```
Comment (efge) [inline nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/QueueImporter.java]: mecanism -> mechanism

File: nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/CQManager.java [hunk 3]
```diff
@@ -1,4 +1,5 @@
-package org.nuxeo.ecm.platform.importer.queue.manager;/*
+package org.nuxeo.ecm.platform.importer.queue.manager;
```
Comment (efge) [inline nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/CQManager.java]: Please put the `package` declaration after the license header (which should be Apache btw).

File: nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/BQManager.java [hunk 4]
```diff
@@ -68,4 +68,14 @@ public void put(int queue, SourceNode node) throws InterruptedException {
...
     }
 
+    @Override
+    public void stop() {
+        // Nothing to do
+    }
+
+    @Override
+    public void init() {
+     // Nothing to do
```
Comment (efge) [inline nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/BQManager.java]: Bad comment indent.

File: nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/QueuesManager.java [hunk 5]
```diff
@@ -62,4 +62,19 @@
...
+
+    /**
+     * Called when the import finishes. Used to clean up things if needed.
+     *
+     * @since 9.1
+     */
+    void stop();
+
+    /**
+     * Called when the import finishes. Used to clean up things if needed.
```
Comment (efge) [inline nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/QueuesManager.java]: Bad Javadoc copy/paste. Also I'd prefer to see `init` before `stop`.

File: nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/CQManager.java [hunk 6]
```diff
@@ -15,43 +16,56 @@
...
 
 /**
  * @since 8.10
  */
 public class CQManager extends AbstractQueuesManager {
 
-    final List<ChronicleQueue> queues;
-    final List<ExcerptAppender> appenders;
-    final List<ExcerptTailer> tailers;
+    List<ChronicleQueue> queues;
```
Comment (efge) [inline nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/CQManager.java]: Why drop the `final` on these? Also don't use package-private declarations, these should be `protected` (to allow use in subclasses).

File: nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/CQManager.java [hunk 7]
```diff
@@ -15,43 +16,56 @@
...
+            throw new RuntimeException("Queue Manager is already setup, stop() it first");
+        }
+
         queues = new ArrayList<>(queuesNb);
         appenders = new ArrayList<>(queuesNb);
         tailers = new ArrayList<>(queuesNb);
 
-        // Create a path for the queue
-        File basePath = new File(System.getProperty("java.io.tmpdir"), "CQ");
+        basePath = new File(System.getProperty("java.io.tmpdir"), "CQ");
```
Comment (efge) [inline nuxeo-importer-queues/src/main/java/org/nuxeo/ecm/platform/importer/queue/manager/CQManager.java]: Please open a ticket to clean up this bad use of fixed-name temporary directories (and check if there are others in this codebase).

**Pull request closed without merging**